tests/unistd/dup2.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <wasi/api.h>
+
+#define __WASI_ERRNO_BADF (UINT16_C(8))
+
+int main() {
```

Comment:
Why write a whole new test here?  Don't we have test overage for dup and dup2 already?

Reply:
This test is called `dup2.c` but it seems to test `dup` as well? 

Can not extend the existing `dup.c` test perhaps?

Reply:
I did this because the existing `dup.c` test had `open` which will be only implemented in the next PR unfortunately.

Reply:
Could we not have modified the existing dup.c tests with and `#ifdef` for WASMFS?

Also the name `dup2.c` is confusing because there happens to be a syscall called `dup2`.

How about the `#ifdef` approach, or calling with `wasmfs_dup.c` and included a `FIXME` to merge it back into the original dup.c test once WASMFS is more complete?